config/disk.go

```diff
@@ -85,36 +88,63 @@ func CloudConfigDirFiles() []string {
 		return []string{}
 	}
 
-	files = util.Filter(files, func(x interface{}) bool {
-		f := x.(os.FileInfo)
-		if f.IsDir() || strings.HasPrefix(f.Name(), ".") {
-			return false
+	var finalFiles []string
+	for _, file := range files {
+		if !file.IsDir() && !strings.HasPrefix(file.Name(), ".") {
+			finalFiles = append(finalFiles, path.Join(CloudConfigDir, file.Name()))
+		}
+	}
+
+	return finalFiles
+}
+
+func applyDebugFlags(rawCfg map[interface{}]interface{}) (map[interface{}]interface{}, error) {
+	cfg := &CloudConfig{}
+	if err := util.Convert(rawCfg, cfg); err != nil {
+		return nil, err
+	}
+
+	if cfg.Rancher.Debug {
+		log.SetLevel(log.DebugLevel)
+		if !util.Contains(cfg.Rancher.Docker.Args, "-D") {
+			cfg.Rancher.Docker.Args = append(cfg.Rancher.Docker.Args, "-D")
+		}
+		if !util.Contains(cfg.Rancher.SystemDocker.Args, "-D") {
+			cfg.Rancher.SystemDocker.Args = append(cfg.Rancher.SystemDocker.Args, "-D")
+		}
+	} else {
+		if util.Contains(cfg.Rancher.Docker.Args, "-D") {
```

Comment:
This should be removed.  This was added because of a side effect.  If rancher.debug is false we shouldn't strip out `-D`.